### PR TITLE
C++: Update documentation for cpp/uncontrolled-allocation-size to clarify its scope

### PR DESCRIPTION
**Pull request by paldepind**
Change the documentation of the `cpp/uncontrolled-allocation-size` query to clarify its intended behavior/scope.

#11215

### CODE REVIEW

File: cpp/ql/test/query-tests/Security/CWE/CWE-190/semmle/TaintedAllocationSize/test.cpp
Comment (paldepind) [inline cpp/ql/test/query-tests/Security/CWE/CWE-190/semmle/TaintedAllocationSize/test.cpp:43]: Changed this comment as this unbounded allocation size is bad per the updated description of the query.

**Review by geoffw0 — COMMENTED** [review]
Changes LGTM.

Can we change the query message as well ("This allocation size is derived from $@ and might overflow.") to put more focus on the tainted / unbounded allocation size than the possibility (which is likely but not certain) of overflow?

**Comment by paldepind** [comment]
Good idea, I missed that. I've updated the query message now as well :smile:

**Review by geoffw0 — COMMENTED** [review]
Looks good to me.

If you add the `ready-for-doc-review` label, the docs team will check over your `.qhelp` and query metadata changes.

You should also add the `no-change-note-required` tag, which will inform CI that you've considered the changes and don't think they require a change note (unless you do, in which case see [this doc](https://github.com/github/codeql/blob/main/docs/change-notes.md) and it goes [here](https://github.com/github/codeql/tree/main/cpp/ql/src/change-notes); roughly speaking, any change that affects query results or the way external developers would write queries of their own should have a change note).

File: cpp/ql/src/Security/CWE/CWE-190/TaintedAllocationSize.qhelp
Comment (subatoi) [inline cpp/ql/src/Security/CWE/CWE-190/TaintedAllocationSize.qhelp:9]:
```suggestion
<p>This code allocates memory using a size value based on user input, with no apparent bound on its magnitude being established. This allows for arbitrary amounts of memory to be allocated.</p>
```

File: cpp/ql/src/Security/CWE/CWE-190/TaintedAllocationSize.qhelp
Comment (subatoi) [inline cpp/ql/src/Security/CWE/CWE-190/TaintedAllocationSize.qhelp:15]:
```suggestion
<p>If the allocation size is calculated by multiplying user input by a
<code>sizeof</code> expression, the multiplication can overflow. When
an integer multiplication overflows in C, the result wraps around and
can be much smaller than intended. A later attempt to write data into
the allocated memory can then be out of bounds.</p>
```

**Review by subatoi — APPROVED** [review]
Looks great, thank you—only some nit comments. Assuming you're happy with the suggestions, feel free to go ahead and merge once you've got an approval from your team (there's no need for me to approve again)

Apologies also if the suggestions mess up some formatting

**Comment by paldepind** [comment]
Thanks @subatoi. I've applied the suggested changes.

**Review by subatoi — APPROVED** [review]
Thank you!

**Review by geoffw0 — APPROVED** [review]
:+1:

**Pull request merged**
